contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -372,12 +375,13 @@ transform_from_ci_as_for_likenode(Node *node, OpExpr *op, like_ilike_info_t like
 		return node;
 	}
 
-	/* Change the opno and oprfuncid to ILIKE */
-	op->opno = like_entry.ilike_oid;
-	op->opfuncid = like_entry.ilike_opfuncid;
-
-	op->inputcollid = tsql_get_oid_from_collidx(collidx_of_cs_as);
-
+	/* Change the opno and oprfuncid to ILIKE if CI collation */
+	if (coll_info_of_inputcollid.collateflags == 0x000f || coll_info_of_inputcollid.collateflags == 0x000d) /* CI */
+	{
+		op->opno = like_entry.ilike_oid;
+		op->opfuncid = like_entry.ilike_opfuncid;
+	}
+	
 	/* 
 	 * This is needed to process CI_AI for Const nodes
```

Comment:
If condition doesnt justify this statement. Either we need to update if condition or improve comment

Reply:
Updated in latest rev